Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -0,0 +1,40 @@
+# Proposal for meta/run.yaml content being present in the composed image
+
+Currently, Capstan runtime support makes use of `$CWD/meta/run.yaml` for composing and running an OSv image. Although this is fine for the composing, it gets tricky for running because `capstan run <IMAGE>` generally does not require user to be inside package root directory.
+
+I propose the following solution:
+
+### Composing image
+File meta/run.yaml is present at package root directory (same as until now)
+
+When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+
+Capstan should create a separate file for each named configuration e.g.
+```
+/osv-runtime
+   |--burek
+   |--pizza   <--------\
+   |                    |
+   |--default-runtime  /
+```
+Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+```
+# content of file /osv-runtime/burek:
+node burek.js
+```
+
+When user runs `capstan package compose` the boot command is set so that the agent is run. A capstan package must be implemented that provides agent funcionality.
```

Comment:
No, `capstan run` remains more or less unchanged!
I just emphasized that images composed with capstan will no longer need to be run by capstan (but by arbitrary infrastructure) **even** if they will use named configurations.

Currently, you need capstan if you want to pick a desired named configuration to **RUN** pre-composed OSv image with. That's because capstan is able to:
* understand your meta/run.yaml
* build run command from it
* set run command to the existing OSv image

Other infrastructures (OpenStack, Amazon) are unable to perform any of this.

The idea is that capstan would prepare everything regarding runtimes in **COMPOSE** time so that standard approaches (e.g. cloud-init) could be used to pick desired runtime in RUN time. So OpenStack would not even need to have capstan installed.